### DISCUSSION TITLE
create backup automatically on container startup

### DISCUSSION THREAD

**Discussion by tob123** [prodrigestivill/docker-postgres-backup-local#140] [Ideas]
Hello,

Can you consider implementing a backup automatically on container startup ?
I know I can execute the ./backup.sh command manually, but removing that manual step makes your container even more comfortable to use..
thank you!

Tob

**Comment by ARezaK** [comment]
this would be great!

**Comment by pandvan** [comment]
Adding this as opt-in would be helpful

**Comment by fritz-net** [comment]
just manually set command/entrypoint when creating the container and this works

**Comment by prodrigestivill** [comment]
Duplicate of #136.

> You already have this option by changing the command to /backup.sh as is explained in the [README.md](https://github.com/prodrigestivill/docker-postgres-backup-local#manual-backups).
> 
> You can override this in k8s, docker-compose or in docker cli as in the example exposed.

**Comment by tob123** [comment]
hi, it is true that the command can be changed. my wish / requirement is two things:
a) container can run on scheduled based backups (without depending on cron); config can be done by environment vars. this closes the gap that docker or docker-compose does not have a scheduler.
b) on top, when the container starts the container can run a backup immediately.
c) put config in one docker-compose file (combined with the db to be backed up for example)

my understanding: if i change the command to backup.sh b),c is covered. if i keep the command as it is a), c) are covered. any suggestion how i can meet a), b), c) ?

regards,

Tobias.

**Comment by prodrigestivill** [comment]
I undestand your proposal, but I can't see much the difference in configuration between adding an environment variable vs changing the command.
Both are a single line in configuration when setting it up, maybe I'm missing some point here.

**Comment by prodrigestivill** [comment]
Implemented in PR #158 it required to also upgrade the go-cron version.

With the new image an initial backup will be launched if `BACKUP_ON_START=TRUE` is set in the `environment`.

Updated README accordingly.
